### PR TITLE
Remove uneeded docblocks

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

Big cleanup (replaces #24931).

Thanks @TomasVotruba for doing all the hard work :)

Before merging, blanck lines between `@param` blocks and `@throws`/`@return`/ ... should be kept.

### CODE REVIEW

**Comment by TomasVotruba** [comment]
@fabpot First version fix for intertag blank lines is here: https://github.com/Symplify/Symplify/pull/599/files#diff-d0813b690623a44a2b83d65d238a5550 in [Symplify v3.2.4](https://github.com/Symplify/BetterReflectionDocBlock/releases/tag/v3.2.4)

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php [hunk 1]
```diff
@@ -36,10 +36,9 @@
     protected $drivers = array();
 
     /**
-     * @param array            $objectManager A configured object manager
-     * @param ContainerBuilder $container     A ContainerBuilder instance
+     * @param array $objectManager A configured object manager
      *
-     * @throws \InvalidArgumentException
+     * @throws InvalidArgumentException
```
Comment (fabpot) [inline src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: @TomasVotruba Not sure why it removed the blackshash here.

**Comment by ostrolucky** [comment]
it also broke phdoc alignments

**Comment by ostrolucky** [comment]
Right. But there might be issue with set priorities. Anyway I'm not blaming your fixer, but pointing to an issue in this PR.

**Comment by TomasVotruba** [comment]
I see. I think @fabpot didn't run it with the other fixer.

@fabpot I recommend running this first, than "@symfony" set from PHP CS Fixer, that's how I use it in practise just in one ECS run.

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php (hunk 1 above)
Comment (TomasVotruba) [reply src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: Fixed: https://github.com/Symplify/Symplify/pull/602

Please try [v3.2.5](https://github.com/Symplify/BetterReflectionDocBlock/releases/tag/v3.2.5)

Comment (fabpot) [reply src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: @TomasVotruba Running the new version give me this error:

```
  Fixing of "src/Symfony/Component/HttpFoundation/Request.php" file by "Symplify\CodingStandard\Fixer\Commenting\RemoveUselessDocBlockFixer" failed: Unknown modifier 'c' in pattern: #@see[\s]+(?<has_
  slash>\\)?https\://tools\.ietf\.org/html/rfc7231#section\-4\.2\.1# in file /Users/fabien/Code/github/Symplify/EasyCodingStandard/vendor/nette/utils/src/Utils/Strings.php on line 604
```

Comment (TomasVotruba) [reply src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: Fixed in 3.2.6

Here is preview https://github.com/symfony/symfony/compare/2.7...TomasVotruba:phpdocs-removal

of:

```yml
# ecs.neon

checkers:
    - PhpCsFixer\Fixer\Phpdoc\PhpdocAlignFixer
    - Symplify\CodingStandard\Fixer\Commenting\RemoveUselessDocBlockFixer

    # remove extra blank lines caused by previous fixer
    - Symplify\CodingStandard\Fixer\Commenting\RemoveEmptyDocBlockFixer
    - Symplify\CodingStandard\Fixer\Commenting\RemoveSuperfluousDocBlockWhitespaceFixer
```

File: src/Symfony/Component/VarDumper/Cloner/Data.php [hunk 2]
```diff
@@ -210,7 +209,6 @@ private function dumpItem($dumper, $cursor, &$refs, $item)
      *
      * @param DumperInterface $dumper
      * @param Cursor          $parentCursor The cursor of the parent hash
-     * @param array           &$refs        A map of all references discovered while dumping
```
Comment (fabpot) [inline src/Symfony/Component/VarDumper/Cloner/Data.php]: @TomasVotruba Checking the diff, I think the current strategy is too aggressive. Here is one such example where the comment seems interesting and should probably be kept. WDYT?

**Comment by fabpot** [comment]
This PR is now ready to be reviewed. I've added a comment on one example where the comment should probably be kept instead of removed (as the description adds some value).

File: src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php (hunk 1 above)
Comment (TomasVotruba) [reply src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php]: I see there are few edge case. I'll try to fix some, but some are very difficult to catch with provided ReflectionDocBlock api.

File: src/Symfony/Component/VarDumper/Cloner/Data.php (hunk 2 above)
Comment (TomasVotruba) [reply src/Symfony/Component/VarDumper/Cloner/Data.php]: @fabpot Thanks for comment. I've noticed this already. I think it's caused by `&` which probably leads to empty comment.

I'm adding to the cases to be fixed

**Comment by TomasVotruba** [comment]
There is 1 last issue reported in fabpot.io: https://fabbot.io/report/symfony/symfony/25859/3f375af281292adbdffb83ae5a6860bc45b393ea

What fixer takes care of that? I'll add it to my comment to complete the setup

File: src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTest.php [hunk 3]
```diff
@@ -98,7 +98,6 @@ public function testGetUserWithEmptyContainer()
     }
 
     /**
-     * @param $token
      *
      * @return ContainerInterface
      */
```
Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Controller/ControllerTest.php]: I would drop the whole, eventually add a `TokenInterface` typehint to `$token` and a return type.

File: src/Symfony/Component/Console/Helper/ProcessHelper.php [hunk 4]
```diff
@@ -27,12 +27,11 @@ class ProcessHelper extends Helper
     /**
      * Runs an external process.
      *
-     * @param OutputInterface      $output    An OutputInterface instance
-     * @param string|array|Process $cmd       An instance of Process or an array of arguments to escape and run or a command to run
-     * @param string|null          $error     An error message that must be displayed if something went wrong
-     * @param callable|null        $callback  A PHP callback to run whenever there is some
-     *                                        output available on STDOUT or STDERR
-     * @param int                  $verbosity The threshold for verbosity
+     * @param string|array|\Process $cmd       An instance of Process or an array of arguments to escape and run or a command to run
```
Comment (chalasr) [inline src/Symfony/Component/Console/Helper/ProcessHelper.php]: wrong `\` on `Process`, class is correctly imported

File: src/Symfony/Component/Console/Helper/ProcessHelper.php [hunk 5]
```diff
@@ -80,8 +79,7 @@ public function run(OutputInterface $output, $cmd, $error = null, $callback = nu
      * This is identical to run() except that an exception is thrown if the process
      * exits with a non-zero exit code.
      *
-     * @param OutputInterface $output   An OutputInterface instance
-     * @param string|Process  $cmd      An instance of Process or a command to run
+     * @param string|\Process $cmd      An instance of Process or a command to run
```
Comment (chalasr) [inline src/Symfony/Component/Console/Helper/ProcessHelper.php]: same

File: src/Symfony/Component/Form/Button.php [hunk 6]
```diff
@@ -122,9 +122,8 @@ public function getParent()
      *
      * This method should not be invoked.
      *
-     * @param int|string|FormInterface $child
-     * @param null                     $type
-     * @param array                    $options
+     * @param int|string|\FormInterface $child
```
Comment (chalasr) [inline src/Symfony/Component/Form/Button.php]: wrong `\`, I stop here :) the fixer seems to consider that classes are all coming from the global namespace

Comment (fabpot) [reply src/Symfony/Component/Form/Button.php]: @TomasVotruba Can you have a look at this issue?

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/Descriptor.php [hunk 7]
```diff
@@ -144,8 +143,7 @@ protected function renderTable(Table $table, $decorated = false)
      * Common options are:
      * * name: name of described service
      *
-     * @param Definition|Alias|object $service
-     * @param array                   $options
+     * @param Definition|\Alias|object $service
```
[the remark this answers is not in the document]
Comment (TomasVotruba) [reply src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/Descriptor.php]: Fixed: https://github.com/symfony/symfony/compare/master...TomasVotruba:phpdos-removal-master#diff-23068eac61c835540fd5aaf1edbd5cd7L129

File: src/Symfony/Component/Console/Helper/ProcessHelper.php (hunk 4 above)
Comment (TomasVotruba) [reply src/Symfony/Component/Console/Helper/ProcessHelper.php]: Fixed: https://github.com/symfony/symfony/compare/master...TomasVotruba:phpdos-removal-master#diff-425e56e3b866cb257f587805f7b2ee54R30

File: src/Symfony/Component/Console/Helper/ProcessHelper.php (hunk 5 above)
Comment (TomasVotruba) [reply src/Symfony/Component/Console/Helper/ProcessHelper.php]: Fixed

File: src/Symfony/Component/Form/Button.php (hunk 6 above)
Comment (TomasVotruba) [reply src/Symfony/Component/Form/Button.php]: Fixed allong with `&$value` references.

I discovered one more error to resolve

**Comment by TomasVotruba** [comment]
Last version to check: https://github.com/symfony/symfony/compare/master...TomasVotruba:phpdos-removal-master

If you comment there, I'll try to fix that asap. Don't mind the indent, just docs removal.

**Comment by chalasr** [comment]
@TomasVotruba Some annotations are replaced by an extra blank line, sometimes an extra blank line is just added between the phpdoc description and the first `@param` annotation:
https://github.com/symfony/symfony/compare/master...TomasVotruba:phpdos-removal-master#diff-87f625b8b8537ac816a1f40590b7ab26R119
https://github.com/symfony/symfony/compare/master...TomasVotruba:phpdos-removal-master#diff-2352a49a1141ef1bd7d22fefa70f22ceR125

**Comment by TomasVotruba** [comment]
@chalasr Oh, I only used the remover fixer. But I've added space cleaner as well. Ready for review :+1:

**Comment by fabpot** [comment]
I've just updated this PR with the new code (thanks @TomasVotruba)

File: src/Symfony/Component/Config/Util/XmlUtils.php [hunk 8]
```diff
@@ -113,18 +113,17 @@ public static function loadFile($file, $schemaOrCallable = null)
      *
      * The following rules applies during the conversion:
      *
-     *  * Each tag is converted to a key value or an array
-     *    if there is more than one "value"
+     * * Each tag is converted to a key value or an array
+     *   if there is more than one "value"
```
Comment (fabpot) [inline src/Symfony/Component/Config/Util/XmlUtils.php]: Not sure about this change. The space is needed to render proper Markdown.

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 9]
```diff
@@ -1413,9 +1401,7 @@ public function dumpParameter($name)
 
     /**
      * @deprecated since version 2.6.2, to be removed in 3.0.
-     *             Use \Symfony\Component\DependencyInjection\ContainerBuilder::addExpressionLanguageProvider instead.
-     *
-     * @param ExpressionFunctionProviderInterface $provider
+     * Use \Symfony\Component\DependencyInjection\ContainerBuilder::addExpressionLanguageProvider instead.
```
Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: I think the previous version is more readable and as it renders the same, I would not change this.

File: src/Symfony/Component/Finder/Tests/Iterator/IteratorTestCase.php [hunk 10]
```diff
@@ -37,14 +37,13 @@ protected function assertOrderedIterator($expected, \Traversable $iterator)
...
      *  array elements.
      *
      *  @param array $expected - an array of arrays. For any two subarrays
-     *      $a and $b such that $a goes before $b in $expected, the method
-     *      asserts that any element of $a goes before any element of $b
-     *      in the sequence generated by $iterator
-     *  @param \Traversable $iterator
+     * $a and $b such that $a goes before $b in $expected, the method
+     * asserts that any element of $a goes before any element of $b
+     * in the sequence generated by $iterator
```
Comment (fabpot) [inline src/Symfony/Component/Finder/Tests/Iterator/IteratorTestCase.php]: And here, I'm pretty sure that the indentation is wrong as this describes the `$expected` param.

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 11]
```diff
@@ -1479,8 +1477,6 @@ public function isMethod($method)
      *
      * @see https://tools.ietf.org/html/rfc7231#section-4.2.1
      *
-     * @param bool $andCacheable Adds the additional condition that the method should be cacheable. True by default.
```
Comment (fabpot) [inline src/Symfony/Component/HttpFoundation/Request.php]: This one should not be removed.

File: src/Symfony/Component/Routing/Matcher/Dumper/DumperCollection.php [hunk 12]
```diff
@@ -47,8 +47,6 @@ public function all()
 
     /**
      * Adds a route or collection.
-     *
-     * @param DumperRoute|DumperCollection The route or collection
```
Comment (fabpot) [inline src/Symfony/Component/Routing/Matcher/Dumper/DumperCollection.php]: Cannot be removed, I think it was removed as it's not valid. `$child` is missing here.

**Comment by fabpot** [comment]
@TomasVotruba Have you had time to have a look at the issues I've noticed here?

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php (hunk 9 above)
Comment (TomasVotruba) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: This is something I can't cover easily. I recommend reverting those manually.

File: src/Symfony/Component/Finder/Tests/Iterator/IteratorTestCase.php (hunk 10 above)
Comment (TomasVotruba) [reply src/Symfony/Component/Finder/Tests/Iterator/IteratorTestCase.php]: This is something I can't cover easily. I recommend reverting those manually.

File: src/Symfony/Component/HttpFoundation/Request.php (hunk 11 above)
Comment (TomasVotruba) [reply src/Symfony/Component/HttpFoundation/Request.php]: I'll look on this one

File: src/Symfony/Component/Routing/Matcher/Dumper/DumperCollection.php (hunk 12 above)
Comment (TomasVotruba) [reply src/Symfony/Component/Routing/Matcher/Dumper/DumperCollection.php]: Yes, this is better to fix manually, since this Fixer relies on valid code.

**Comment by TomasVotruba** [comment]
@fabpot I missed your comments here, sorry. I'm looking into it.
I'm able to fix 1 issue, the rest you'll have to revert manually.

File: src/Symfony/Component/HttpFoundation/Request.php (hunk 11 above)
Comment (TomasVotruba) [reply src/Symfony/Component/HttpFoundation/Request.php]: Oh, I see it's commented bellow. This Fixer is not able to differentiate between commented argument and extra useless docs for param.

Just revert this manually

**Comment by nicolas-grekas** [comment]
@TomasVotruba would you like to take over doing the manual tweaks? Otherwise I'd suggest to close.

**Comment by TomasVotruba** [comment]
@nicolas-grekas Sure. I can get to in till Friday.

**Comment by TomasVotruba** [comment]
@nicolas-grekas  I'll try to re-send this. Which branch should I target now?

**Comment by nicolas-grekas** [comment]
3.4 is the oldest maintained branch now.

**Comment by TomasVotruba** [comment]
@nicolas-grekas Kk

**Pull request closed without merging**